torch/_inductor/fx_utils.py

```diff
@@ -192,21 +238,20 @@ def any_user_may_alias(node):
 
             return False
 
-        def should_process_node(node):
-            # node.target for nodes returning true from this function
-            # are called under fake mode and does not work for inductor
-            # lowerings. We check if the node.target is an aten operator
-            # or operator.getitem which is used when returning multiple
-            # tensors from an op.
-            return node.op == "call_function" and (
-                isinstance(node.target, torch._ops.OpOverload)
-                or node.target == operator.getitem
-                or node.target
-                == torch._inductor.fx_passes.reinplace._generalized_scatter
+        def should_process_node(node: torch.fx.Node) -> bool:
+            # node.target will called with FakeTensor arguments, which is not supported
+            # by Inductor lowerings
+            return callable(node.target) and not hasattr(
+                node.target, "_inductor_lowering_function"
             )
 
+        # Since subgraph I/O semantics are assumed not to change, it's safe to
+        # unconditionally update them first.
+        for updater in self.subgraph_updaters.values():
+            updater.incremental_update()
```

Comment:
question:

If I have a subgraph that looks like:
```
def subgraph(x):
   y = x.view(-1) # C
   return y.sin()
```
and it's being used in the outer graph:
```
def graph(x):
    y = x.view(-1)  # A
    return invoke_subgraph(subgraph, y) # B
```   
and then I delete the meta["val"] on `A`,

what happens?

I think we need to update Node A and Node B (those need new FakeTensors). But do we need to update node C as well?